### ISSUE TITLE
Dotfiles are not moved/copied into public/

### ISSUE THREAD

**Issue by edmorley** [cloudfoundry-incubator/staticfile-buildpack#19]
As part of the compile, files in $root_dir are moved into public/.

There are a few issues:
1) |mv| doesn't include dotfiles/dot-directories, so we end up with some files present in public/ and some left behind. I believe we should move/copy _all_ files, since (a) that seems more predictable, and (b) people can already exclude files using their manifest (and in fact the default ignore-files excludes dotfiles anyway).
2) The message says "Copying", but we use |mv|. We should change one or the other.

**Comment by jchesterpivotal** [comment]
Worth comparing with the behaviour of PHP buildpack, which has a similar "move and filter" strategy. In that case it introduces a lot of complexity because other parts of the buildpack have to be aware of the relocation.

**Comment by edmorley** [comment]
My use case is:

```yaml
name: test
instances: 1
mem: 64M
framework:
  type: buildpack
env:
  BUILDPACK_URL: git://github.com/cloudfoundry-incubator/staticfile-buildpack.git
  FORCE_HTTPS: true
hooks:
  pre-staging:
  - git clone --depth 1 https://github.com/foo/bar.git src
cron: 
  - "* * * * * cd $HOME/public/src && git pull"
```
With a Staticfile containing:
```yaml
root: src
```

...to have a simple static files site auto-update from Github.

Unfortunately at the moment, the .git directory isn't copied to public/, so the |git pull| of the cron fails. In addition, since the working directory of the repo ($HOME/src) was |mv|ed, I need to reset the working directory, git pull, then rsync the files to $HOME/public/src - which is all a bit of a dance just to get a simple site live. (And yeah I know for many people they wouldn't want the .git directory in public, but it really doesn't matter for this use case).

**Comment by jtarchie** [comment]
@edmorley, you appear to be using quite a bit Stackato features, which are not apart of the standard [cf-release](http://github.com/cloudfoundry/cf-release). I don't know how well this work flow would work with the static-buildpack and don't have environment to test against.

If your looking for a static site from a github repo, why not look at [Github pages](https://pages.github.com/). They have support for quite a few features, including custom domains.

**Comment by edmorley** [comment]
Yeah I would much rather be able to use Github pages for this - using Mozilla's Stackato instance is overkill - however you cannot force HTTPS on Github pages sadly (more context at https://bugzilla.mozilla.org/show_bug.cgi?id=1050477#c11).

That said, the comment 0 behaviour seems unexpected (my use case aside) - would you agree? :-)

**Comment by edmorley** [comment]
What would you feel about changing:
```bash
status "Copying project files into public/"
mkdir -p $cache_dir/public
mv $root_dir/* $cache_dir/public
mv $cache_dir/public .
```
To:
```bash
status "Copying project files into public/"
mkdir -p $cache_dir/public
cp $root_dir $cache_dir/public
mv $cache_dir/public .
```

ie: Actually copying rather than moving (to match the status message), and copying the directory instead of just the non-dotfiles within?

**Comment by edmorley** [comment]
The php buildpack (at least from a quick glance) seems to copy dotfiles:
https://github.com/cloudfoundry/php-buildpack/blob/3ba3832ef476a881be6970fff1671698c8afdae8/lib/compile_helpers.py#L37

It also specifically excludes manifest.yml, which would also fix issue #20.
Should we just switch to using rsync instead of mv here?
https://github.com/cloudfoundry-incubator/staticfile-buildpack/blob/caa6cea1527f4bc8720ee566ff1c4f99331746c1/bin/compile#L35

It would save the mv of Staticfile back from public/ to app_root as well:
https://github.com/cloudfoundry-incubator/staticfile-buildpack/blob/caa6cea1527f4bc8720ee566ff1c4f99331746c1/bin/compile#L42
